common/src/main/java/me/shedaniel/lightoverlay/common/LightOverlayTicker.java

```diff
@@ -263,7 +266,7 @@ public byte getCrossType(BlockPos pos, Holder<Biome> biome, BlockPos down, Block
         // Check block state allow spawning (excludes bedrock and barriers automatically)
         if (!blockBelowState.isValidSpawn(world, down, TESTING_ENTITY_TYPE.get()))
             return LightOverlay.CROSS_NONE;
-        if (!LightOverlay.mushroom && Biome.BiomeCategory.MUSHROOM == Biome.getBiomeCategory(biome))
+        if (!LightOverlay.mushroom && biome.is(Biomes.MUSHROOM_FIELDS))
```

Comment:
This should be `biome.is(BiomeTags.MUSHROOM)` (if that doesn't exist, it should check `c:mushroom` and `forge:is_mushroom`

Reply:
See here: https://github.com/itsmeow/IMDLib/blob/1.19-architectury/fabric/src/main/java/dev/itsmeow/imdlib/entity/util/fabric/BiomeTypesImpl.java

Reply:
Even if `c:mushroom` and `forge:is_mushroom` points to `BiomeTags.MUSHROOM` ?
https://github.com/FabricMC/fabric/blob/1.19/fabric-convention-tags-v1/src/generated/resources/data/c/tags/worldgen/biome/mushroom.json#L4
https://github.com/MinecraftForge/MinecraftForge/blob/1.19.x/src/generated/resources/data/forge/tags/worldgen/biome/is_mushroom.json#L3

Reply:
the point is mods can add mushroom biomes that are not just mushroom fields, you are only checking for mushroom fields

Reply:
I don't really know how it works, but is it better like that? :
```java
final TagKey<Biome> mushroom = TagKey.create(Registry.BIOME_REGISTRY, new ResourceLocation("mushroom"));
        final TagKey<Biome> is_mushroom = TagKey.create(Registry.BIOME_REGISTRY, new ResourceLocation("is_mushroom"));
        if (!LightOverlay.mushroom && biome.tags().anyMatch(tag -> tag.equals(mushroom) || tag.equals(is_mushroom)))
            return LightOverlay.CROSS_NONE;
```

Reply:
I don't think you can do anyMatch and equals. You need to use is. Also the resourcelocation is not namespaced for c and forge.

Reply:
I'm not an expert developer with Java, I don't really how to make cross-forge-fabric code like that because the code is design for fabric and forge at the same time

Reply:
tags are common code, literally just put new ResourceLocation("forge", "is_mushroom")

Reply:
Like that
```java
final TagKey<Biome> mushroom = TagKey.create(Registry.BIOME_REGISTRY, new ResourceLocation("c", "mushroom"));
final TagKey<Biome> is_mushroom = TagKey.create(Registry.BIOME_REGISTRY, new ResourceLocation("forge", "is_mushroom"));
if (!LightOverlay.mushroom && (biome.is(mushroom) || biome.is(is_mushroom)))
    return LightOverlay.CROSS_NONE;
```